packages/pyright-internal/src/languageServerBase.ts

```diff
@@ -1235,13 +1233,12 @@ export abstract class LanguageServerBase implements LanguageServerInterface {
         return MarkupKind.PlainText;
     }
 
-    private async _getProgressReporter(
-        progressId: string | number | undefined,
-        reporter: WorkDoneProgressReporter,
-        title: string,
-        token: CancellationToken
-    ) {
-        if (progressId) {
+    private async _getProgressReporter(reporter: WorkDoneProgressReporter, title: string, token: CancellationToken) {
+        // This is a bit ugly, but we need to determine whether the provided reporter
+        // is an actual client-side progress reporter or a dummy (null) progress reporter
+        // created by the LSP library. If it's the latter, we'll create a server-initiated
+        // progress reporter.
+        if (reporter.constructor.name !== 'NullProgressReporter') {
```

Comment:
Oh, that's a good point. Can you think of any other ways for us to do this check? If not, we may need to wait for a change in the LSP library.

Reply:
The best hack I can come up with is to call `attachWorkDone` to get a `NullProgressReporter`, then compare the constructors for identity.

```ts
const nullProgressReporter = attachWorkDone(undefined as any, undefined);

// ...
private async _getProgressReporter(reporter: WorkDoneProgressReporter, title: string, token: CancellationToken) {
    if (reporter.constructor !== nullProgressReporter.constructor) { .... }
}
```

Reply:
That's clever. Very hacky, but clever.

Reply:
Oop, no, that doesn't work. 

```
> class Foo {}
undefined
> Foo.constructor === Foo.constructor
true
> class Bar {}
undefined
> Foo.constructor === Bar.constructor
true
```

Reply:
No, I'm silly. Those are the class's constructors, which are of course equal because that's `object`.

```
> (new Foo()).constructor === (new Foo()).constructor
true
> (new Foo()).constructor === (new Bar()).constructor
false
```

The hack should work. Sorry for the noise.